lib/dial/dial.go

```diff
@@ -0,0 +1,37 @@
+package dial
+
+import (
+	"fmt"
+
+	"github.com/tarantool/go-tarantool/v2"
+)
+
+// Dial creates new dialer according to the options.
+func New(opts Opts) (tarantool.Dialer, error) {
```

Comment:
We could add unit-tests here. Something like:

```Go
func TestNew(t *testing.T) {
    cases := []struct{
        name     string
        opts        dial.Opts
        expected tarantool.DIaler
    } {
        {
            Name: "simple",
            Opts: dial.Opts{foo},
            expected: tarantool.NetDialer{foo},
        }
        ...
    }
    
    for _, tc := range cases {
        dialer, err := dial.New(tc.opts)

        require.NoError(t, err)
        require.Equal(t, tc.expected, dialer)
    } 
}
```

For simplicity we can test this only with ssl enabled. But you can make two different test files. See
https://github.com/tarantool/go-tarantool/blob/461ad486865e0f401a46568a688ec0851b2aa020/ssl_test.go#L1-L2